src/cmap/conn/mod.rs

```diff
@@ -46,9 +46,15 @@ pub struct ConnectionInfo {
     pub id: u32,
 
     /// A server-generated identifier that uniquely identifies the connection. Available on server
-    /// versions 4.2+. This may be used to correlate driver connections with server logs.
+    /// versions 4.2+. This may be used to correlate driver connections with server logs. This
+    /// could be a truncated value based on input server_id.
```

Comment:
Nit: how about "If the connection ID sent by the server is too large for an i32, this will be a truncated value." (or "will be None." if we go the other way) for wording?  I think it's useful for our users when we can be more precise about the behavior.